test/coordinate/cartesian_differential.cpp

```diff
@@ -0,0 +1,117 @@
+#define BOOST_TEST_MODULE cartesian_differential_test
+
+#include <boost/test/unit_test.hpp>
+#include <boost/units/quantity.hpp>
+#include <boost/units/systems/si/plane_angle.hpp>
+#include <boost/units/systems/si/prefixes.hpp>
+#include <boost/units/systems/si/length.hpp>
+#include <boost/units/systems/si/velocity.hpp>
+#include <boost/units/systems/si/time.hpp>
+#include <boost/units/systems/angle/degrees.hpp>
+#include <boost/astronomy/coordinate/arithmetic.hpp>
+#include <boost/astronomy/coordinate/differential.hpp>
+
+using namespace std;
+using namespace boost::astronomy::coordinate;
+using namespace boost::units::si;
+using namespace boost::geometry;
+using namespace boost::units;
+namespace bud = boost::units::degree;
+
+BOOST_AUTO_TEST_SUITE(cartesian_differential_constructor)
+
+BOOST_AUTO_TEST_CASE(cartesian)
+{
+    //checking construction from value
+    auto motion1 = make_cartesian_differential
+    (1.5*meters/seconds, 9.0*si::kilo*meters/seconds, 3.0*meters/seconds);
+    BOOST_CHECK_CLOSE(motion1.get_dx().value(), 1.5, 0.001);
+    BOOST_CHECK_CLOSE(motion1.get_dy().value(), 9.0, 0.001);
+    BOOST_CHECK_CLOSE(motion1.get_dz().value(), 3, 0.001);
+    
+    //copy constructor
+    auto motion2 = make_cartesian_differential(motion1);
+    BOOST_CHECK_CLOSE(motion1.get_dx().value(), motion2.get_dx().value(), 0.001);
+    BOOST_CHECK_CLOSE(motion1.get_dy().value(), motion2.get_dy().value(), 0.001);
+    BOOST_CHECK_CLOSE(motion1.get_dz().value(), motion2.get_dz().value(), 0.001);
+    
```

Comment:
Can you please add tests for checking if the quantity stored in each coordinate is as expected.
you can use [`BOOST_TEST`](https://www.boost.org/doc/libs/1_70_0/libs/test/doc/html/boost_test/testing_tools/boost_test_universal_macro.html)
you can do something like this:
```
BOOST_TEST(std::is_same<decltype(motion1.get_dx()), bu::quantity<bu::divide_typeof_helper<bu::si::length, bu::si::time>::type>>::value)
```

There can be many ways of doing this maybe even more efficient too it's up to you. Add such tests wherever you feel is needed in differential and representation tests especially when any kind of conversions are performed.